packages/react/test/specs/components/Dropdown/Dropdown-test.tsx

```diff
@@ -104,854 +116,804 @@ describe('Dropdown', () => {
-        .simulate('keydown', { keyCode: keyboardKey.Enter, key: 'Enter' })
+    it('it takes the value of the default prop but can be changed', () => {
+      const defaultValue = items[2]
+      const itemToBeClickedIndex = 1
+      const { triggerButton, clickOnItemAtIndex } = renderDropdown({
+        defaultValue,
+        defaultOpen: true,
+      })
 
-      expect(onSelectedChange).toHaveBeenCalledTimes(1)
-      expect(onSelectedChange).toHaveBeenCalledWith(
-        null,
-        expect.objectContaining({
-          value: items[0],
-        }),
-      )
+      expect(triggerButton).toHaveTextContent(defaultValue)
+
+      clickOnItemAtIndex(itemToBeClickedIndex)
+
+      expect(triggerButton).toHaveTextContent(items[itemToBeClickedIndex])
     })
 
-    it('is set by using Tab on highlighted item', () => {
+    it('has onSelectedChange called when item is added', () => {
+      const itemToClickIndex = 2
       const onSelectedChange = jest.fn()
-      const wrapper = mountWithProvider(
-        <Dropdown onSelectedChange={onSelectedChange} items={items} />,
-      )
-      const triggerButton = getTriggerButtonWrapper(wrapper)
+      const { clickOnItemAtIndex } = renderDropdown({ open: true, onSelectedChange })
 
-      triggerButton.simulate('click')
-      const itemsList = getItemsListWrapper(wrapper)
-      itemsList
-        .simulate('keydown', { keyCode: keyboardKey.ArrowDown, key: 'ArrowDown' })
-        .simulate('keydown', { keyCode: keyboardKey.Tab, key: 'Tab' })
+      clickOnItemAtIndex(itemToClickIndex)
 
       expect(onSelectedChange).toHaveBeenCalledTimes(1)
       expect(onSelectedChange).toHaveBeenCalledWith(
         null,
         expect.objectContaining({
-          value: items[0],
+          value: items[itemToClickIndex],
+          searchQuery: items[itemToClickIndex],
         }),
       )
     })
 
-    it('is set by using Shift+Tab on highlighted item', () => {
+    it('has onValueChanged called when item is added or removed on multiple', () => {
+      // it will actually be the third, since one is already removed from the list due to defaultValue.
+      const itemToClickIndex = 2
+      const defaultSelectedItemIndex = 1
       const onSelectedChange = jest.fn()
-      const wrapper = mountWithProvider(
-        <Dropdown onSelectedChange={onSelectedChange} items={items} />,
-      )
-      const triggerButton = getTriggerButtonWrapper(wrapper)
+      const { clickOnItemAtIndex, keyDownOnSelectedItemAtIndex } = renderDropdown({
+        open: true,
+        defaultValue: [items[defaultSelectedItemIndex]],
+        onSelectedChange,
+        multiple: true,
+      })
 
-      triggerButton.simulate('click')
-      const itemsList = getItemsListWrapper(wrapper)
-      itemsList
-        .simulate('keydown', { keyCode: keyboardKey.ArrowDown, key: 'ArrowDown' })
-        .simulate('keydown', { keyCode: keyboardKey.Tab, key: 'Tab', shiftKey: true })
+      clickOnItemAtIndex(itemToClickIndex - 1)
```

Comment:
```tsx
const { clickOnItemAtIndex, keyDownOnSelectedItemAtIndex } = renderDropdown({
  items: ['foo', 'bar', 'baz'],
  defaultValue: ['bar'],
})
//
clickOnItemAtIndex(1)
```

This looks confusing, it will be easier to undestand with out magic numbers and magic values. It applies actually for all test cases. May be it will be more hard to refactor, but...

Reply:
I will still need to add explaining comments even if we use duplicated numbers. Not sure it will get things any better.